### DISCUSSION TITLE
ChangePropertyEventData can cause a crash if component properties are different between client/server

### DISCUSSION THREAD

**Discussion by Ydrec** [FakeFishGames/Barotrauma#15393] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

While modding via lua i was sending bunch of networking ChangePropertyEventData events. Started reproducably crashing so i dug deeper. 

After logging every property sent and received i found that for docking port same property index in networkmessage was producing a different property. Wires have property [width](https://github.com/FakeFishGames/Barotrauma/blob/38ca721cb0fd94aadbb9c1886ba2338bbd5f1180/Barotrauma/BarotraumaClient/ClientSource/Items/Components/Signal/Wire.cs#L110) on client which doesnt exist on server and it was shifting property index by one and causing [ReadPropertyChange](https://github.com/FakeFishGames/Barotrauma/blob/38ca721cb0fd94aadbb9c1886ba2338bbd5f1180/Barotrauma/BarotraumaShared/SharedSource/Items/Item.cs#L3760) to read a string from message which was containing a bool. 

Seems like that function assumes that [allProperties](https://github.com/FakeFishGames/Barotrauma/blob/38ca721cb0fd94aadbb9c1886ba2338bbd5f1180/Barotrauma/BarotraumaShared/SharedSource/Items/Item.cs#L3764) list returned by GetProperties<Editable>() is exactly the same without checking

![sshot-3](https://github.com/user-attachments/assets/83644225-1147-475b-93ed-5270103c5e9f)

### Reproduction steps

Start a server with lua
Load into submarine
enter following command in console with immediate crash following
`lua LuaUserData.RegisterType('Barotrauma.Character+CharacterStatusEventData')  CharacterStatusEventData = LuaUserData.CreateStatic('Barotrauma.Character+CharacterStatusEventData', true) for item in Item.ItemList do if tostring(item.Prefab.Identifier) == 'dockinghatch' then connectionpanel = item.GetComponent(Components.ConnectionPanel) connectionpanel.Locked = not connectionpanel.Locked property = connectionpanel.SerializableProperties[Identifier('Locked')] Networking.CreateEntityEvent(item, Item.ChangePropertyEventData(property, connectionpanel)) break end end`

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.7.7.0 (Winter Update)

### Which operating system did you encounter this bug on?

Windows

### Relevant error messages and crash reports

```shell
[Vanilla] Error while reading a message from server. 
Entity: Docking Hatch (ID: 1649)
Exception thrown while reading a message of the type "EntityEvent" at position 32.
Inner exception: Tried to read too much data from segment.
   at Barotrauma.Networking.SegmentTableReader`1.SegmentReadMsg.Check() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Utils/SegmentTable.cs:line 162
   at Barotrauma.Networking.SegmentTableReader`1.SegmentReadMsg.ReadString() in <DEV>/Barotrauma/BarotraumaShared/SharedSource/Utils/SegmentTable.cs:line 201
   at Barotrauma.Item.ReadPropertyChange_Patch2(Item this, IReadMessage msg, Boolean inGameEditableOnly, Client sender)
   at Barotrauma.Item.ClientEventRead(IReadMessage msg, Single sendingTime) in <DEV>/Barotrauma/BarotraumaClient/ClientSource/Items/Item.cs:line 2022
   at Barotrauma.Networking.ClientEntityEventManager.Read(ServerNetSegment type, IReadMessage msg, Single sendingTime) in <DEV>/Barotrauma/BarotraumaClient/ClientSource/Networking/NetEntityEvent/ClientEntityEventManager.cs:line 235
```

**Comment by Rokvach** [comment]
Fixed and tested the fix to be working correctly. Fix coming in the next update. Closing discussion.
